collect_app/src/main/java/org/odk/collect/android/instancemanagement/InstanceSubmitter.java

```diff
@@ -147,19 +149,11 @@ public InstanceSubmitter(Analytics analytics) {
      *
      * @param isAutoSendAppSettingEnabled whether the auto-send option is enabled at the app level
      */
-    public static boolean formShouldBeAutoSent(String jrFormId, boolean isAutoSendAppSettingEnabled) {
-        Cursor cursor = new FormsDao().getFormsCursorForFormId(jrFormId);
-        String formLevelAutoSend = null;
-        if (cursor != null && cursor.moveToFirst()) {
-            try {
-                int autoSendColumnIndex = cursor.getColumnIndex(AUTO_SEND);
-                formLevelAutoSend = cursor.getString(autoSendColumnIndex);
-            } finally {
-                cursor.close();
-            }
+    public static boolean shouldFormBeSent(FormsRepository formsRepository, String jrFormId, String jrFormVersion, boolean isAutoSendAppSettingEnabled) {
```

Comment:
In my head this would ideally be a `private` method that is just an implementation detail of `InstanceSubmitter`. The `InstanceSubmitter` would then have tests around it that setup instances and check that they are sent in the right contexts. Looking at though there is a lot of dependencies on static calls we'd need to pull out to write that test (`Collect.getInsance().getComponent()` and `GeneralSharedPreferences.isAutoSendEnabled()` for example). This PR moves the needle forward so I don't think it makes sense to block it on that rework.

Could you maybe just add a `@Deprecated` to this method and a `@deprecated` Javadoc explaining that really it should be a private method and that that requires refactoring `InstanceSubmitter` to make it testable/modular?